### PR TITLE
[5.5] Add a firstWhere method to the Collection class

### PR DESCRIPTION
**Pull request by JosephSilber**
```php
$firstActiveUser = $users->firstWhere('active', true);
```

### CODE REVIEW

File: tests/Support/SupportCollectionTest.php [hunk 1]
```diff
@@ -45,6 +45,19 @@ public function testFirstWithDefaultAndWithoutCallback()
         $this->assertEquals('default', $result);
     }
 
+    public function testFirstWehre()
```
Comment (Dylan-DPC) [inline tests/Support/SupportCollectionTest.php]: typo in function name :stuck_out_tongue:

**Comment by taylorotwell** [comment]
Just needed this yesterday :smile:

**Comment by jordyvandomselaar** [comment]
Where's firstOrFailWhere?

**Comment by JosephSilber** [comment]
@jordyvandomselaar right here:

```php
$value = $c->firstWhere(...) ?? throw new Exception(...);
```

**Comment by jordyvandomselaar** [comment]
No no no, if we're adding methods for the heck of it that provide no value anyways we're gonna need this.

**Pull request merged**